services/blockchain-indexer/shared/indexer/transactionProcessor/dpos/voteDelegate.js

```diff
@@ -13,36 +13,110 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
+const BluebirdPromise = require('bluebird');
+
 const {
 	Logger,
 	MySQL: { getTableInstance },
 } = require('lisk-service-framework');
+
+const {
+	getBase32AddressFromPublicKey,
+	getBase32AddressFromHex,
+} = require('../../../utils/accountUtils');
 const config = require('../../../../config');
 
 const logger = Logger();
 
 const MYSQL_ENDPOINT = config.endpoints.mysql;
-const transactionsIndexSchema = require('../../../database/schema/transactions');
+const votesIndexSchema = require('../../../database/schema/votes');
 
-const getTransactionsIndex = () => getTableInstance('transactions', transactionsIndexSchema, MYSQL_ENDPOINT);
+const getVotesIndex = () => getTableInstance('votes', votesIndexSchema, MYSQL_ENDPOINT);
 
 // Command specific constants
 const commandID = 1;
 const commandName = 'voteDelegate';
 
+const getVoteIndexingInfo = async (tx) => {
+	const votes = tx.params.votes.map(async vote => {
+		const voteEntry = {};
+
+		voteEntry.sentAddress = getBase32AddressFromPublicKey(tx.senderPublicKey);
+		voteEntry.receivedAddress = getBase32AddressFromHex(vote.delegateAddress);
+		voteEntry.amount = vote.amount;
+		return voteEntry;
+	});
+	return votes;
+};
+
+const indexVoteTrx = async (vote, trx) => {
+	const votesDB = await getVotesIndex();
+
+	const incrementParam = {
+		increment: {
+			amount: BigInt(vote.amount),
+		},
+		where: {
+			sentAddress: vote.sentAddress,
+			receivedAddress: vote.receivedAddress,
+		},
+	};
+
+	const numRowsAffected = await votesDB.increment(incrementParam, trx);
+	if (numRowsAffected === 0) {
+		await votesDB.upsert(vote.voteObject, trx);
```

Comment:
`getVoteIndexingInfo` doesn't return any `vote.voteObject` property. Please fix it.